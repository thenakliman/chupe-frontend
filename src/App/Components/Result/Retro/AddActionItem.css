.add-action-item-button {
    padding: 0.8rem 1rem;
    color: white;
    cursor: pointer;
    margin: 2rem;
}

.add-action-item-container {
    display: flex;
    flex-direction: column;
    background: #fff;
    padding: 4rem 3rem;
    border: 1px solid #ccc;
    top: 11rem;
    left: 41rem;
    position: fixed;
    width: 35rem;
}

.action-item-due-date {
  width: 10rem;
  margin-bottom: 3rem;
  padding: 0.5rem;
}

.action-item-assigned-to {
  width: 10rem;
  margin-bottom: 1rem;
  padding: 0.5rem;
}

.add-action-item-creation-input {
    padding: 0.8rem 0;
    width: 100%;
    font-size: 1.4rem;
    margin: 2rem 0;
    height: 6rem;
}