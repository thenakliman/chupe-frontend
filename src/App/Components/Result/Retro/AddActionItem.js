import React from 'react';
import propTypes from 'prop-types';

require('./AddActionItem.css');

export class AddActionItem extends React.Component {
    constructor(props) {
      super(props);
      this.state = {description: '', assignedTo: '', deadlineToAct: ''};
      this.handleRetroDescription = this.handleRetroDescription.bind(this);
      this.handleActionItemAssignedToChange = this.handleActionItemAssignedToChange.bind(this);
      this.handleDueDateChange = this.handleDueDateChange.bind(this);
      this.createRetroActionItem = this.createRetroActionItem.bind(this);
    }

    handleRetroDescription(description) {
      this.setState({description: description});
    }

    handleActionItemAssignedToChange(assignedTo) {
      this.setState({assignedTo: assignedTo});
    }

    handleDueDateChange(deadlineToAct) {
      this.setState({deadlineToAct: deadlineToAct});
    }

    createRetroActionItem() {
      this.props.createActionItem(this.state);
      this.setState({description: '', assignedTo: '', deadlineToAct: ''});
    }

    render() {
      return (
      <div className='retro-pop-up-container'>
        <div id='add-action-item-container-id'
             className='add-action-item-container'>
                Description
                <textarea className={'add-action-item-creation-input'}
                       id={'add-action-item-description'}
                       maxLength={2000}
                       onChange={(event) =>
                          this.handleRetroDescription(event.target.value)}
                       value={this.state.description}
                />
                Due Date
                <input className={'action-item-due-date'}
                       id={'action-item-due-date'}
                       onChange={(event) =>
                          this.handleDueDateChange(event.target.value)}
                       value={this.state.deadlineToAct}
                       type={'date'}
                />
                Assigned To
                <select value={this.state.assignedTo}
                        className={'action-item-assigned-to'}
                        onChange={(event) =>
                            this.handleActionItemAssignedToChange(event.target.value)}>
                  <option>Select user</option>
                  {
                    this.props.users.map((user) =>
                      <option key={user.userName}
                              value={user.userName}>
                           {user.userName}
                      </option>
                    )
                  }
                </select>
                <div className='add-action-item-buttons-container'>
                    <button id='add-action-item-button-id'
                      className='add-action-item-button'
                      type='button'
                      onClick={this.createRetroActionItem}
                    >
                      Add Action Item
                    </button>
                    <button id='cancel-add-action-item-button-id'
                      className='add-action-item-button'
                      onClick={this.props.closeAddActionItemPopUp}
                      type='button'
                    >
                      Cancel
                    </button>
                </div>
        </div>
      </div>);
    }
}

AddActionItem.propTypes = {
  closeAddActionItemPopUp: propTypes.func.isRequired,
  createActionItem: propTypes.func.isRequired,
  users: propTypes.array.isRequired,
};
