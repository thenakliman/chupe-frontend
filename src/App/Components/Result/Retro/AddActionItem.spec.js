/* eslint-disable */
import React from 'react';
import {AddActionItem} from './AddActionItem';
/* eslint-enable */
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

describe('Show create action item pop up component', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(
      <AddActionItem
        users={[{userName: 'user-1'}]}
        createActionItem={jest.fn()}
        closeAddActionItemPopUp={jest.fn()}
      />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should have initial state for description, assigned to and due date', () => {
    const wrapper = shallow(
      <AddActionItem
        users={[{userName: 'user-1'}]}
        createActionItem={jest.fn()}
        closeAddActionItemPopUp={jest.fn()}
      />);
    expect(wrapper.state()).toEqual({
      description: '',
      assignedTo: '',
      deadlineToAct: '',
    });
  });

  it('should call closeAddActionItemPopUp on click of cancel button', () => {
    const closeAddActionItemPopUp = jest.fn();
    const wrapper = shallow(
      <AddActionItem
        users={[{userName: 'user-1'}]}
        createActionItem={jest.fn()}
        closeAddActionItemPopUp={closeAddActionItemPopUp}
      />);

    wrapper.find('#cancel-add-action-item-button-id').simulate('click');
    expect(closeAddActionItemPopUp).toHaveBeenCalledWith();
  });

  it('should change state description state on edit', () => {
    const wrapper = shallow(
      <AddActionItem
        users={[{userName: 'user-1'}]}
        createActionItem={jest.fn()}
        closeAddActionItemPopUp={jest.fn()}
      />);

    const description = 'description';
    wrapper.find('#add-action-item-description').simulate('change',
     {target: {value: description}});

    expect(wrapper.state().description).toEqual(description);
  });

  it('should change state of assignedTo state on edit', () => {
    const wrapper = shallow(
      <AddActionItem
        users={[{userName: 'user-1'}]}
        createActionItem={jest.fn()}
        closeAddActionItemPopUp={jest.fn()}
      />);

    const description = 'description';
    wrapper.find('.action-item-assigned-to').simulate('change',
     {target: {value: 'user'}});

    expect(wrapper.state().assignedTo).toEqual('user');
  });

  it('should change state of due date state on edit', () => {
    const wrapper = shallow(
      <AddActionItem
        users={[{userName: 'user-1'}]}
        createActionItem={jest.fn()}
        closeAddActionItemPopUp={jest.fn()}
      />);

    const description = 'description';
    wrapper.find('#action-item-due-date').simulate('change',
     {target: {value: '2019/12/12'}});

    expect(wrapper.state().deadlineToAct).toEqual('2019/12/12');
  });

  it('should value change state to initial on click of add action item', () => {
    const wrapper = shallow(
      <AddActionItem
        users={[{userName: 'user-1'}]}
        createActionItem={jest.fn()}
        closeAddActionItemPopUp={jest.fn()}
      />);

    wrapper.find('#add-action-item-button-id').simulate('click');

    expect(wrapper.state()).toEqual({
      description: '',
      assignedTo: '',
      deadlineToAct: '',
    });
  });

  it('should call retro point create on click of add retro point', () => {
    const createActionItem = jest.fn();
    const wrapper = shallow(
      <AddActionItem
         users={[{userName: 'user-1'}]}
        createActionItem={createActionItem}
        closeAddActionItemPopUp={jest.fn()}
      />);

    wrapper.find('#add-action-item-button-id').simulate('click');

    expect(createActionItem).toHaveBeenCalledWith({
        description: '',
        assignedTo: '',
        deadlineToAct: '',
      });
  });
});
