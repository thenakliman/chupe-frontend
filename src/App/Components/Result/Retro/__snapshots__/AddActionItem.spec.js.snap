// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Show create action item pop up component should match snapshot 1`] = `
<div
  className="retro-pop-up-container"
>
  <div
    className="add-action-item-container"
    id="add-action-item-container-id"
  >
    Description
    <textarea
      className="add-action-item-creation-input"
      id="add-action-item-description"
      maxLength={2000}
      onChange={[Function]}
      value=""
    />
    Due Date
    <input
      className="action-item-due-date"
      id="action-item-due-date"
      onChange={[Function]}
      type="date"
      value=""
    />
    Assigned To
    <select
      className="action-item-assigned-to"
      onChange={[Function]}
      value=""
    >
      <option>
        Select user
      </option>
      <option
        key="user-1"
        value="user-1"
      >
        user-1
      </option>
    </select>
    <div
      className="add-action-item-buttons-container"
    >
      <button
        className="add-action-item-button"
        id="add-action-item-button-id"
        onClick={[Function]}
        type="button"
      >
        Add Action Item
      </button>
      <button
        className="add-action-item-button"
        id="cancel-add-action-item-button-id"
        onClick={[MockFunction]}
        type="button"
      >
        Cancel
      </button>
    </div>
  </div>
</div>
`;
